src/libraries/System.Private.CoreLib/src/System/Numerics/INumberBase.cs

```diff
@@ -49,15 +49,17 @@ public interface INumberBase<TSelf>
         /// <exception cref="OverflowException"><paramref name="value" /> is not representable by <typeparamref name="TSelf" />.</exception>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         static virtual TSelf CreateChecked<TOther>(TOther value)
+#nullable disable
             where TOther : INumberBase<TOther>
+#nullable restore
         {
             TSelf? result;
 
             if (typeof(TOther) == typeof(TSelf))
             {
                 result = (TSelf)(object)value;
             }
-            else if (!TSelf.TryConvertFromChecked(value, out result) && !TOther.TryConvertToChecked(value, out result))
+            else if (!TSelf.TryConvertFromChecked(value, out result) && !TOther.TryConvertToChecked<TSelf>(value, out result))
```

Comment:
Why is this extra type annotation needed now?

Reply:
Without it, type inference infers the type parameter to be `TSelf?` rather than `TSelf`, as it's basing it off the type of the `result` field.